### ISSUE TITLE
Recent commit breaks router composition, backward compatibility

### ISSUE THREAD

**Issue by agurtovoy** [pedronauck/micro-router#24]
#19 (merged on July 18) makes it impossible to combine routers, e.g.

auth.js:
```
export default router(
    post( "/auth/request-token", requestToken ),
    ...
);
```

root.js:
```
import auth from "./auth";

export default router(
    get( "/hello", helo ),
    auth
);
```
This used to work, we had to fork and revert to keep our app working.

**Comment by pedronauck** [comment]
Closed on [release v3.0.0](https://github.com/pedronauck/micro-router/releases/tag/v3.0.0)

**Comment by davidhund** [comment]
Hi @pedronauck — Is route composition supposed to work with `^3.0.0`?

The [tests pass](https://github.com/pedronauck/micro-router/blob/master/lib/index.test.js#L62) but I can't get it to work with the following:

```
// index.js (`micro-dev index.js`)
const { router, get } = require('microrouter');

const fooBarRoute = router(
	get('/foo', () => `/foo route`),
	get('/bar', () => `/bar route`)
);

const bazRoute = router(
	get('/baz', () => `/baz route`)
);

module.exports = router(
	fooBarRoute,  // <-- /foo and /bar work
	bazRoute      // <-- /baz fails
);
```

Both `/foo` and `/bar` work, but `/baz` => `[404] Cannot GET /baz`...
